### PR TITLE
feat: add AI conversion with a new method + minore fixies

### LINKED ISSUE #14: PYPI: update readme and docs to include video processing
As per title we need to update the readme on the pypi page

### LINKED ISSUE #15: Example video appears to be corrupt when viewed in Firefox
The example video in the documentation (on this page, not PyPI) appears as corrupted on Firefox 123.0, on both desktop (linux) and mobile (android 13) versions of the browser:

Video can't be played because the file is corrupt

This seems to be a compatibility issue with Firefox (or maybe the Gecko engine) in particular, as the video appears and plays correctly on Chrome 122.0. I don't have easy access to Safari right now to see how a WebKit-based browser handles it.

### LINKED ISSUE #19: Implement a new method for converting an image with PaletteNet

### LINKED ISSUE #21: Change license to GNU AFFERO GENERAL V3

### PR DESCRIPTION
**Pull request by TheJoin95**
This PR will introduce a new conversion method using the PaletteNet implementation in a six input color palette.

It's our first try to jump in a conversion - recolor palette algorithm.
It's been a long journey since the first iteration here. Back in 2020 we thought of implementing a GAN but it was too hard for us to implementing it.

Nowadays the tools are super powerful and easy to use (pytoch 2.2 it's crazy).
Thans to the researchers who made the PaletteNet, to the guys who implemented it: it enable us to make this PR.

It's not perfect and I think the linear algorithm it's better in some results BUT the AI model is working great for the wallpaper: it's giving back a really high definition image (no data loss, pixel quality).

Within this PR come also some minor fixies such as video in readme fix and others.

We're going to have a major release on this.

Closes #21 
Closes #19
Closes #15
Closes #14 

As a demo we're going to add the new method in our demo website and API on @Schrodinger-Hat/imagegonord

### CODE REVIEW

File: ImageGoNord/GoNord.py [hunk 1]
```diff
@@ -417,7 +425,70 @@ def converted_loop(self, is_rgba, pixels, original_pixels, maxRow, maxCol, minRo
...
+        for hex, rgb_value in self.PALETTE_DATA.items():
+            a = []
+            for j in [2,4,6]:
+                a.append(int(hex[j-2:j],16))
+            palette.append(a)
+
+        try:
+            pal_np = np.array(palette).reshape(1,6,3)/255
+        except:
+            print("You have too many colors in your palette for the model. Auto reducing the array. It's limited at 6 colors, now you have: ", len(palette))
```
Comment (Wabri) [inline ImageGoNord/GoNord.py:475]: I dont get the message inside the print, do you want to comunicate that the only color that will be taken will be the first 6? It is clear in line 476, but can we be a more explicit?

```suggestion
            print("You have too many colors in your palette for the model, this feature is limited to 6 colours, now you have: ", len(palette), "! I'll take the first 6!")
```

Or something like this.

Maybe we can also throw an error without proceed further and just print the reason in the error.

File: ImageGoNord/GoNord.py [hunk 2]
```diff
@@ -439,20 +516,24 @@ def convert_image(self, image, save_path='', parallel_threading=False):
...
-            for row in range(step, image.size[0] + step, step):
-                args = (is_rgba, pixels, original_pixels, row, image.size[1], row - step, 0)
-                t = threading.Thread(target=self.converted_loop, args=args)
-                t.daemon = True
-                t.start()
-                threads.append(t)
-
-            for t in threads:
-                t.join(timeout=30)
+            if (parallel_threading == False):
```
Comment (Wabri) [inline ImageGoNord/GoNord.py:523]:
```suggestion
            if not parallel_threading:
```

File: ImageGoNord/GoNord.py [hunk 3]
```diff
@@ -439,20 +516,24 @@ def convert_image(self, image, save_path='', parallel_threading=False):
         original_image.close()
         pixels = self.load_pixel_image(image)
         is_rgba = (image.mode == 'RGBA')
-        if (parallel_threading == False):
-            self.converted_loop(is_rgba, pixels, original_pixels, image.size[0], image.size[1])
+
+        if (use_model):
```
Comment (Wabri) [inline ImageGoNord/GoNord.py:520]:
```suggestion
        if use_model:
```

File: ImageGoNord/GoNord.py [hunk 4]
```diff
@@ -439,20 +516,24 @@ def convert_image(self, image, save_path='', parallel_threading=False):
...
+                    args = (is_rgba, pixels, original_pixels, row, image.size[1], row - step, 0)
+                    t = threading.Thread(target=self.converted_loop, args=args)
+                    t.daemon = True
+                    t.start()
+                    threads.append(t)
+
+                for t in threads:
+                    t.join(timeout=30)
 
         if (self.USE_GAUSSIAN_BLUR == True):
```
Comment (Wabri) [inline ImageGoNord/GoNord.py:538]: I know that is not in the scope of this pr but:
```suggestion
        if self.USE_GAUSSIAN_BLUR:
```

**Review by Wabri — CHANGES_REQUESTED** [review]
Some minor fixes and I'm confident in your skill in creating the model.py

File: ImageGoNord/GoNord.py (hunk 4 above)
Comment (TheJoin95) [reply ImageGoNord/GoNord.py:538]: Nice catch

**Comment by TheJoin95** [comment]
@Wabri please re-review :)

**Review by Wabri — APPROVED** [review]
LGTM 🥇 Lets go to the moon!

**Comment by TheJoin95** [comment]
Let's gooooo !

**Pull request merged**